share/completions/zig.fish

```diff
@@ -0,0 +1,282 @@
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l import-memory -d "Import memory from the environment (Wasm)"
+complete -x -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l initial-memory -d "Initial size of the linear memory (Wasm)"
+complete -x -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l max-memory -d "Maximum size of the linear memory (Wasm)"
+complete -x -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l global-base -d "Where to start to place global data (Wasm)"
+complete -x -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l export -d "Force a symbol to be exported (Wasm)"
+
+## Test options
+complete -x -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l test-filter -d "Skip tests that do not match filter"
+complete -x -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l test-name-prefix -d "Add prefix to all tests"
+complete -x -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l test-cmd -d "Specify test execution command"
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l test-cmd-bin -d "Appends test binary path to `--test-cmd`"
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l test-evented-io -d "Runs the test in evented I/O mode"
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l test-no-exec -d "Compiles test binary without running it"
+
+## Debug options
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -o ftime-report -d "Print timing diagnostics"
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -o fstack-report -d "Print stack size diagnostics"
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l verbose-link -d "Display linker invocations"
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l verbose-cc -d "Display C compiler invocations"
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l verbose-air -d "Enable compiler debug output for Zig AIR"
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l verbose-mir -d "Enable compiler debug output for Zig MIR"
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l verbose-llvm-ir -d "Enable compiler debug output for LLVM IR"
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l verbose-cimport -d "Enable compiler debug output for C imports"
+complete -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l verbose-llvm-cpu-features -d "Enable compiler debug output for LLVM CPU features"
+complete -x -c zig -n "__fish_seen_subcommand_from $zig_build_generic_commands" -l debug-log -d "Enable printing debug/info log messages for scope"
```

Comment:
unfortunately this disables file completions, can we add something to the `-n` parameter so it doesn't?
I believe that clang completions only trigger if the current token starts with `-`, so maybe that?

Looks really great otherwise!

Reply:
The argument of `--debug-log` is not files (<https://github.com/ziglang/zig/blob/370793a36b3de34ead8456553a2aa2c56f0d3de8/src/main.zig#L497>). So, this is as expected.

I'll fix the file completion for `zig cc` and `zig c++` to be enabled.